core/test_network/src/lib.rs

```diff
@@ -656,6 +772,13 @@ impl PeerBuilder {
         peer
     }
 
+    /// Create and start a peer with preapplied arguments at the specified port.
+    pub async fn start_with_port(self, port: u16) -> Peer {
```

Comment:
It seems you can add `port` into `PeerBuilder` like this:
```rust
pub struct PeerBuilder {
    configuration: Option<Configuration>,
    genesis: WithGenesis,
    // ...
    port: Option<u16>, // <===
}
```
and also add separate setter `with_port(self, port: u16) -> Self` into `PeerBuilder`

Reply:
Much prefer the syntax `port(self, port: u16) -> Self`.

Reply:
It felt to me that `with_port` goes more in line with the other builder setters.